File: defibrillators.py
import sys
from math import cos, pi, pow, sqrt

def to_f(input: str) -> float:
    return float(input.replace(',', '.'))

def to_rad(deg: float) -> float:
    return deg * pi / 180

def distance(lat_a, lon_a, lat_b, lon_b) -> float:
    x = (lon_b - lon_a) * cos((lat_a + lat_b)/2)
    y = lat_b - lat_a
    return sqrt(pow(x, 2) + pow(y, 2)) * 6371

lon = to_rad(to_f(input()))
lat = to_rad(to_f(input()))
n = int(input())
min_d = None
address = None
for i in range(n):
    defib = input()
    defib_elts = defib.split(';')
    lat_d = to_rad(to_f(defib_elts.pop()))
    lon_d = to_rad(to_f(defib_elts.pop()))
    dist = distance(lat, lon, lat_d, lon_d)
    if (min_d is None) or (dist < min_d):
        min_d = dist
        address = defib_elts[1]

print(address)

File: defibrillators.swift
import Glibc
import Foundation

public struct StderrOutputStream: TextOutputStream {
  public mutating func write(_ string: String) { fputs(string, stderr) }
}
public var errStream = StderrOutputStream()

extension Float {
  var rad: Float { self * .pi / 180 }
}

func toFloat(_ input: String) -> Float {
  Float(input.replacingOccurrences(of: ",", with: "."))!
}

func distance(latA: Float, lonA: Float, latB: Float, lonB: Float) -> Float {
  let x = (lonB - lonA) * cos((latA + latB)/2)
  let y = latB - latA
  return sqrtf(pow(x, 2) + pow(y, 2)) * 6371
}

let lon = toFloat(readLine()!).rad
let lat = toFloat(readLine()!).rad
let n = Int(readLine()!)!
var minDist: Float?
var address: String?
if n > 0 {
  for i in 0...(n-1) {
    var defib = readLine()!.split(separator: ";").map(String.init)
    let latD = toFloat(defib.popLast()!).rad
    let lonD = toFloat(defib.popLast()!).rad
    let dist = distance(latA: lat, lonA: lon, latB: latD, lonB: lonD)
    if minDist == nil || dist < minDist! {
      minDist = dist
      address = defib[1]
    }
  }
}

print(address!)